thunder/__init__.py

```diff
@@ -203,8 +203,18 @@ def __init__(self, model, compiled_model_call):
         super().__init__()
         self._model = model
 
+        del self.training
```

Comment:
at glance this reads a tad aggressive. why and how would this be needed?

Reply:
I'll add a comment. This is to automatically take the `self._model.training` instead of `self.training`.